app/controllers/feedbacks_controller.rb

```diff
@@ -0,0 +1,74 @@
+class FeedbacksController < ApplicationController
+  before_action :set_feedback, only: [:show, :edit, :update, :destroy]
+  before_action :set_feedbackable, only: [:new, :create]
+
+  def index
+    authorize Feedback
+    @feedbacks = policy_scope(Feedback)
+    if params[:group_offer_id]
+      @feedbacks = @feedbacks.where(feedbackable_id: params[:group_offer_id])
+      @volunteer = current_user.volunteer || @feedbacks.first.volunteer
+    elsif params[:assignment_id]
+      @feedbacks = @feedbacks.where(feedbackable_id: params[:assignment_id])
+      @volunteer = Assignment.find(params[:assignment_id]).volunteer
+    elsif params[:volunteer_id]
+      @feedbacks = @feedbacks.where(volunteer_id: params[:volunteer_id])
+      @volunteer = Volunteer.find(params[:volunteer_id])
+    end
+  end
+
+  def show; end
+
+  def new
+    @feedback = Feedback.new(feedbackable: @feedbackable, volunteer: @volunteer,
+      author: current_user)
+    authorize @feedback
+  end
+
+  def edit; end
+
+  def create
+    @feedback = Feedback.new(feedback_params.merge(author_id: current_user.id))
+    authorize @feedback
+    if @feedback.save
+      redirect_to url_for([@feedback.volunteer, @feedback]), make_notice
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @feedback.update(feedback_params)
+      redirect_to url_for([@feedback.volunteer, @feedback]), make_notice
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @feedback.destroy
+    redirect_back(fallback_location: url_for(@feedbackable))
+  end
+
+  private
+
+  def set_feedbackable
+    @feedbackable = Assignment.find(params[:assignment_id]) if params[:assignment_id]
+    @feedbackable = GroupOffer.find(params[:group_offer_id]) if params[:group_offer_id]
+    return @volunteer = @feedbackable.volunteer if @feedbackable.class == Assignment
+    @volunteer = current_user.volunteer if current_user.volunteer?
+    @volunteer = Volunteer.find(params[:volunteer]) if params[:volunteer]
```

Comment:
Same concern here, but maybe I'm just missing something :)

Reply:
Its because if a superadmin wants to leave a feedback on a GroupOffer, there is no volunteer available. Thats why I passed it thru a get param in the new Feedback link. To avoid overlappings with :volunteer_id, i chose :volunteer
I find it strange that a superadmin can and should leave feedback, but AFIR superadmins are supposed to be able